server/server.go

```diff
@@ -835,14 +836,31 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(noRouteResponseCode int) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(noRouteResponseCode)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(responseCode int) http.Handler {
+	if responseCode == 0 {
+		responseCode = http.StatusNotFound
+	}
+	if http.StatusText(responseCode) == "" {
+		log.Warnf("Wrong NoRouteResponseCode %d configured. Falling back to 404 (http.StatusNotFound)", responseCode)
```

Comment:
Another point: How about just letting `newNotFounderHandler` return a `(http.Handler, error)` tuple and return a non-nil error for this case? It'd then be up to the caller to handle it accordingly (i.e., fall back to 404) and not be nested inside the handler factory.